src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,787 @@
+
+    tripletList.push_back({rowOffset + 1,
+                           columnOffset,
+                           16 * (1 / std::pow(poly[knotIndex - 1].duration, 3)
+                                 + 1 / std::pow(poly[knotIndex].duration, 3))});
+
+    tripletList.push_back({rowOffset + 1,
+                           columnOffset + 1,
+                           3 * (-1 / std::pow(poly[knotIndex - 1].duration, 2)
+                                + 1 / std::pow(poly[knotIndex].duration, 2))});
+}
+
+void QuinticSpline::Impl::addTripletPreviousKnot(const int& knotIndex,
+                                                 const int& rowOffset,
+                                                 const int& columnOffset,
+                                                 std::vector<Eigen::Triplet<double>>& tripletList)
+{
+    const auto& poly = polynomials;
+
+    // The following triplets represent this matrix
+    // /    8          1    \
+    // | ------     ------  |
+    // |  2                 |
+    // | T           T      |
+    // |  i - 1       i - 1 |
+    // |                    |
+    // |                    |
+    // |   14          2    |
+    // | ------     ------  |
+    // |  3          2      |
+    // | T          T       |
+    // \  i - 1      i - 1  /
+
+    tripletList.push_back({rowOffset, columnOffset, 8 / std::pow(poly[knotIndex - 1].duration, 2)});
+    tripletList.push_back({rowOffset, columnOffset + 1, 1 / poly[knotIndex - 1].duration});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset, 14 / std::pow(poly[knotIndex - 1].duration, 3)});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset + 1, 2 / std::pow(poly[knotIndex - 1].duration, 2)});
+}
+
+void QuinticSpline::Impl::addTripletNextKnot(const int& knotIndex,
+                                             const int& rowOffset,
+                                             const int& columnOffset,
+                                             std::vector<Eigen::Triplet<double>>& tripletList)
+{
+    const auto& poly = polynomials;
+
+    // The following triplets represent this matrix
+    // /    8          1    \
+    // | - ---        ---   |
+    // |    2               |
+    // |   T           T    |
+    // |    i           i   |
+    // |                    |
+    // |                    |
+    // |  14           2    |
+    // |  --         - ---  |
+    // |   3            2   |
+    // |  T            T    |
+    // \   i            i   /
+
+    tripletList.push_back({rowOffset, columnOffset, -8 / std::pow(poly[knotIndex].duration, 2)});
+    tripletList.push_back({rowOffset, columnOffset + 1, 1 / poly[knotIndex].duration});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset, 14 / std::pow(poly[knotIndex].duration, 3)});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset + 1, -2 / std::pow(poly[knotIndex].duration, 2)});
```

Comment:
According to your derivation in https://github.com/dic-iit/element_dcm-walking/issues/188#issuecomment-673542645,  `-2 / std::pow(poly[knotIndex].duration, 2)`  must be raised to the power 3 `-2 / std::pow(poly[knotIndex].duration, 3)`. There seems to be a typo somewhere (most possibly in the issue comment). Please cross-verify.

Reply:
There was a typo in the issue comment. Fixed! Thanks for pointing out.